javaparser-core/src/main/java/com/github/javaparser/resolution/declarations/AssociableToAST.java

```diff
@@ -52,7 +51,23 @@
      * In these cases getWrappedNode is particularly nice because it returns the right type of AST node,
      * not just a Node.
      */
-    default Optional<N> toAst() {
-        throw new UnsupportedOperationException();
+    Optional<Node> toAst();
```

Comment:
Maybe a default implementation that returns Optional.empty() would simplify the code